### DISCUSSION TITLE
How to pass AWSSessionCredentialsProvider in aws-sdk-java-v2

### DISCUSSION THREAD

**Discussion by dreamerHarshit** [aws/aws-sdk-java-v2#5402] [Q&A] [answered]
### Describe the bug

In the earlier version of aws java SDK, we used to connect with aws s3 using the following code snippet

`AmazonS3EncryptionClientV2Builder.standard().
   .withCredentials(<Object of AWSSessionCredentialsProvider>) 
   .withEncryptionMaterialsProvider()
   .withKmsClient()
   .withEndpointConfiguration().build`

### Expected Behavior

As it used to support AWSSessionCredentialsProvider in earlier versions so was expecting it to work with new version as well
or if there is any alternative of this?

### Current Behavior

Getting error that credentialProvider() in aws-sdk-java-v2 does not take AwsSessionCredentialsProvider as input

### Reproduction Steps

NA

### AWS Java SDK version used

2.25.10

### JDK version used

java17

### Operating System and version

rhel8

**Comment by bhoradc** [comment, marked as answer]
Hi @dreamerHarshit, 

There is no equivalent of v1 `AWSSessionCredentialsProvider` in v2. But you will find all the implementing classes of `AWSSessionCredentialsProvider` implemented in v2 as well. Kindly refer to [AwsCredentialsProvider](https://sdk.amazonaws.com/java/api/latest/software/amazon/awssdk/auth/credentials/AwsCredentialsProvider.html) for more details. 

Regards, 
Chaitanya
